pyfar/dsp/dsp.py

```diff
@@ -2669,3 +2669,175 @@ def normalize(signal, reference_method='max', domain='auto',
+    is realized in the frequency domain using the corresponding spectra
+    :math:`X_1[k]` and :math:`X_2[k]`
+
+    .. math:: c = \mathrm{IFFT\{X_1 X_2^-\}}
+
+    with the inverse fourier transform denoted by IFFT and :math:`X^-` denoting
+    the spectrum of a time-reversed and complex conjugated time signal, i.e.,
+    ``X_minus = fft(conj(x[::-1]))`` where the conjugate is required if the
+    input has complex-valued time data.
+
+    Parameters
+    ----------
+    signal_1 : Signal
+        The first input signal. It must have the same sample rate as
+        `signal_2` and its ``cshape`` must be broadcastable to that of
+        `signal_2`.
+    signal_2 : Signal
+        The second input signal. It must have the same sample rate as
+        `signal_1` and its ``cshape`` must be broadcastable to that of
+        `signal_1`.
+    mode : str, optional
+
+        ``'full'``
+            Computes the full correlation by zero padding the input signals
+            to a length of ``signal_1.n_samples + signal_2.n_samples - 1``
+            before applying the Fourier transform (see equations above).
+        ``'cyclic'``
+            Computes the cyclic correlation, which uses the input signals as
+            they are. In this case `signal_1` and `signal_2` must have the
+            same number of samples (length).
+
+        The default is ``'full'``.
+
+    Returns
+    -------
+    correlation : TimeData
+        The correlation :math:`c[l]` is contained in ``correlation.time`` and
+        the lags in seconds, i.e., the delays applied to `signal_2`
+        (see equations above) are contained in ``correlation.times``.
+        ``correlation.time`` is complex if one of the input signals has
+        complex-valued time data. The cshape of `correlation` matches the
+        cshape to which `signal_1` and `signal_2` were broadcasted. The lags
+        can be converted to samples by multiplication with
+        ``signal_1.sampling_rate``. In this case, they are in the interval
+        ``[-signal_2.n_samples + 1, signal_1.n_samples - 1]`` if the full
+        correlation was computed. In case of the cyclic correlation, they are
+        in the interval
+        ``[-(signal_1.n_samples // 2) + 1, signal_1.n_samples // 2]``
+        if the signals have an even number of samples, and in the interval
+        ``[-(signal_1.n_samples // 2), signal_1.n_samples // 2]``
+        if the signals have an odd number of samples.
+
+    Examples
+    --------
+
+    Linear and cyclic auto-correlation of a perfect sequence. Perfect sequences
+    have unit auto-correlation for :math:`l=0` and zero auto-correlation
+    otherwise
+
+    .. plot::
+
+        >>> import pyfar as pf
+        >>>
+        >>> signal = pf.signals.linear_perfect_sweep(2**8)
+        >>>
+        >>> for mode in ['full', 'cyclic']:
+        >>>     cor = pf.dsp.correlate(signal, signal, mode)
+        >>>     ax = pf.plot.time(cor / pf.dsp.energy(signal), label=mode)
```

Comment:
```suggestion
        >>> for mode, ls in zip(['full', 'cyclic'], ['-', ':']):
        >>>     cor = pf.dsp.correlate(signal, signal, mode)
        >>>     ax = pf.plot.time(cor / pf.dsp.energy(signal), label=mode, linestyle=ls)
```
It's a bit hard to see what's happening at t=0.